### PR TITLE
MATAD02 - Add tracklist to product metafields

### PR DESCRIPTION
**Pull request by ivancosmin**
### What does this PR change? 
Sent tracklist metafield from csv import

### Relevant Task Links 
https://app.asana.com/0/1207979806654377/1208134552685391

### CODE REVIEW

File: app/Console/Commands/PushProductsMetafields.php [hunk 1]
```diff
@@ -70,9 +71,32 @@ public function handle(): void
...
+                    ->get();
+
+                $tracklist = [];
+                foreach ($tracks as $track) {
+                    $tracklist[] = [
+                        'title' => $track->track_title,
+                        'track_number' => $track->track_index,
+                        'duration' => $track->duration
+                    ];
+                }
```
Comment (radmen) [inline app/Console/Commands/PushProductsMetafields.php:88]: :large_blue_circle: You could use `map()` and avoid using `foreach`

```suggestion
                $tracklist = DB::table('imported_tracklist')
                    ->where('release_code', str_replace('-', '', $migrationOriginData['sku']))
                    ->get()
                    ->map(fn (\stdClass $item) => [
                            'title' => $item->track_title,
                            'track_number' => $item->track_index,
                            'duration' => $item->duration
                    ])
                    ->all();
```

File: app/Console/Commands/PushProductsMetafields.php [hunk 2]
```diff
@@ -70,9 +71,32 @@ public function handle(): void
...
+
+                $metafields[] = [
+                    'key' => 'tracklist',
+                    'namespace' => 'beggars',
+                    'value' => $tracklist,
```
Comment (radmen) [inline app/Console/Commands/PushProductsMetafields.php:93]: :red_circle: You need to `json_encode()` the value. AFAIR Shopify will reject any complex type, even though this is supposed to be `json`. It has to be already serialized.

File: app/Console/Commands/PushProductsMetafields.php [hunk 3]
```diff
@@ -70,9 +71,32 @@ public function handle(): void
                             'type' => 'date',
                             'ownerId' => $productData['admin_graphql_api_id'],
                         ]
-                    ]);
+                    ];
                 }
 
+                $tracks = DB::table('imported_tracklist')
+                    ->where('release_code', str_replace('-', '', $migrationOriginData['sku']))
+                    ->get();
```
Comment (radmen) [inline app/Console/Commands/PushProductsMetafields.php:79]: :thinking: When is this one imported?

**Review by radmen — CHANGES_REQUESTED** [review]
I have finished reviewing your Pull Request. I have flagged my comments with the following emojis:

- 🔴 indicates that changes are required
- 🔵 suggests making a change
- 💬 is a tip, general feedback, or a comment
- 🤔 is a question
  
Please review my comments and make the necessary changes. If I left any code suggestions, they might not work. Check them first before making any commits. 

If you have any questions or concerns, please let me know.

File: app/Console/Commands/PushProductsMetafields.php (hunk 3 above)
Comment (ivancosmin) [reply app/Console/Commands/PushProductsMetafields.php:79]: i will make a command to populate from csv tablse: imported_products, imported_variants, imported_tracklist
i will make it with path argument to be universal command

File: app/Console/Commands/PushProductsMetafields.php (hunk 1 above)
Comment (ivancosmin) [reply app/Console/Commands/PushProductsMetafields.php]: fixed

File: app/Console/Commands/PushProductsMetafields.php (hunk 2 above)
Comment (ivancosmin) [reply app/Console/Commands/PushProductsMetafields.php]: fixed

**Review by radmen — APPROVED** [review]

**Pull request merged**